emukit/model_wrappers/gpy_model_wrappers.py

```diff
@@ -134,7 +134,10 @@ def generate_hyperparameters_samples(self, n_samples=20, n_burnin=100, subsample
 
         """
         self.model.optimize(max_iters=self.n_restarts)
-        self.model.param_array[:] = self.model.param_array * (1. + np.random.randn(self.model.param_array.size) * 0.01)
+        unfixed_params = [param for param in self.model.flattened_parameters if not param.is_fixed]
+        for param in unfixed_params:
+            # Add jitter by multiplying with log-normal noise with mean 1 and standard deviation 0.01 
+            param *= np.random.lognormal(np.log(1. / np.sqrt(1.0001)), np.sqrt(np.log(1.0001)))
```

Comment:
Can this comment be expanded? Right now it just says what the code does, which isn't particuarly helpful. Instead it should ideally talk about why jitter is needed, and why it has to be done with log-normal noise instead of simple uniform.

The original code wasn't documented at all, but that was poor, so let's use the opporunity to improve here

Reply:
To be perfectly honest, I am not entirely certain about why the jitter was needed in the first place.

I'm guessing that it's because the parameters are optimised first, which means that the gradient with respect to posterior probability will be close to `0`, and this would be a bad initialisation for HMC as it utilises the gradients.

I'm happy to add a comment saying this if my interpretation is correct.

Reply:
As for why I changed it to a log-normal, this is essentially to ensure the sign of the parameter value isn't flipped (in case it's constrained to be positive).

Reply:
Both replies make sense to me, feel free to add them as comments

Reply:
oh, you arealdy did. cool, thanks!